framework/include/meshgenerators/MeshGenerator.h

```diff
@@ -203,6 +210,31 @@ MeshGenerator::declareMeshProperty(const std::string & data_name, const T & init
   return data;
 }
 
+template <typename T>
+T &
+MeshGenerator::setMeshProperty(const std::string & data_name, const T & data_value)
+{
+  if (!_app.executingMeshGenerators())
+    mooseError(
+        "Updating mesh meta data cannot occur in the constructor of mesh generators");
```

Comment:
why?

Reply:
I was thinking we impose this criteria to make it clear that setMeshGenerator should only be used outside of the constructor, otherwise if you want to update the value in the constructor just use declareMeshGenerator? If you don't think this is necessary, then I can remove this condition

Reply:
yeah let s remove. We dont really want to add a unit test to get that error tested as well